wfe/wfe.go

```diff
@@ -722,23 +722,31 @@ func (wfe *WebFrontEndImpl) UpdateAccount(
 		Contact []string `json:"contact"`
 		Status  string   `json:"status,omitempty"`
 	}
-	err := json.Unmarshal(postData.body, &updateAcctReq)
-	if err != nil {
-		wfe.sendError(
-			acme.MalformedProblem("Error unmarshaling account update JSON body"), response)
-		return
-	}
-
-	existingAcct, prob := wfe.getAcctByKey(postData.jwk)
-	if prob != nil {
-		wfe.sendError(prob, response)
-		return
+	var existingAcct *core.Account
+	if postData.postAsGet {
+		existingAcct, prob = wfe.validPOSTAsGET(postData)
+		if prob != nil {
+			wfe.sendError(prob, response)
+			return
+		}
+	} else {
+		err := json.Unmarshal(postData.body, &updateAcctReq)
+		if err != nil {
+			wfe.sendError(
+				acme.MalformedProblem("Error unmarshaling account update JSON body"), response)
+			return
+		}
+		existingAcct, prob = wfe.getAcctByKey(postData.jwk)
+		if prob != nil {
+			wfe.sendError(prob, response)
+			return
+		}
 	}
 
 	// if this update contains no contacts or deactivated status,
 	// simply return the existing account and return early.
 	if updateAcctReq.Contact == nil && updateAcctReq.Status != acme.StatusDeactivated {
```

Comment:
I think it would be sensible to return an error if we reach the "retrieve an account because there was no update body" case but `wfe.strict == true && postData.postAsGet == false` because this indicates the client is using the old way of querying account information but wants Pebble to be strict about that sort of thing.

Does that make sense?

Reply:
I was thinking about that myself, but decided against it, because the draft doesn't say one should not retrieve the account data this way. From how I understand it, this is still a valid way, even though it is not explicitly advertised any more. I'm kind of torn, so I let you decide :)

Reply:
IMO: If the draft doesn't describe fetching the account that way then it shouldn't be supported. I think `-strict` should reject deprecated behaviour.

Reply:
I've added a commit which does that. I've used `acme.MalformedProblem()`; or should I use a different error class instead?

Reply:
`acme.MalformedProblem()` sounds correct to me. Thanks @felixfontein !